### PR TITLE
chore: add documentation for trobleshooting

### PR DESCRIPTION
**Pull request by raksiv**
A few simple guides to document how a user might recover from an inconsistent state during early development.

### CODE REVIEW

File: docs/troubleshooting/pulumi.mdx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## Troubleshooting a deploment failure
```
Comment (tjholm) [inline docs/troubleshooting/pulumi.mdx:1]: deployment

File: docs/troubleshooting/updating-nitric.mdx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## Updating Nitric
+
+If you're ready to upgrade your projects to a newer version of Nitric
+
+- Start by removing any versions of the membrane which are currently residing on Docker `docker system prune -a`
```
Comment (tjholm) [inline docs/troubleshooting/updating-nitric.mdx:5]: A full prune may be a bit gratuitous, we might be able to get them to pull the latest version of the membrane explicitly using docker pull?

We're thinking once we have more stable versioning we will actually tie template versions to membrane versions as well, if it's explicitely specified it could save them the hassle of having to remove stale membranes.

File: docs/tutorials/go/stripe-integration.mdx [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+
+Next we'll be asked if we want an initial service in the project and what template the service will use.
+For this project we will want a service. This service will handle requests for greeting and respond to the greetings. For this we will use the `nodets12` template.
+
+x
+? Include an example service?
+function/nodets12
+function/nodejs12
+function/python37
+function/java11
```
Comment (tjholm) [inline docs/tutorials/go/stripe-integration.mdx:29]: Could we move this tutorial to a separate PR?

The troubleshooting and tutorials are independently valuable.

**Review by tjholm — CHANGES_REQUESTED** [review]

File: docs/tutorials/go/stripe-integration.mdx (hunk 3 above)
Comment (raksiv) [reply docs/tutorials/go/stripe-integration.mdx:29]: Yup, I'll separate it - slipped in

File: docs/troubleshooting/pulumi.mdx (hunk 1 above)
Comment (raksiv) [reply docs/troubleshooting/pulumi.mdx]: Updated.

File: docs/troubleshooting/updating-nitric.mdx (hunk 2 above)
Comment (raksiv) [reply docs/troubleshooting/updating-nitric.mdx:5]: Sure  - I am happy to add this option in - 

- docker image rm nitricimages/membrane-local
- checkout membrane
- run make dev-docker or make aws-docker
- docker pull nitricimages/membrane-local

Anything missing?

Comment (raksiv) [reply docs/troubleshooting/updating-nitric.mdx]: Added steps describing alternative to pruning.

File: docs/troubleshooting/updating-nitric.mdx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+## Updating Nitric
+
+If you're ready to upgrade your projects to a newer version of Nitric
+
+- Start by refreshing any versions of the membrane which are currently residing on Docker:
+  - Run `docker image rm nitricimages/membrane-local`
+  - Checkout the latest version of the membrane
+  - Run `make dev-docker` (or other target e.g. `make aws-docker`. Please refer to the [membrane readme](https://github.com/nitrictech/membrane/blob/main/README.md) for more details
+  - Update the local membrane with `docker pull nitricimages/membrane-local`
```
Comment (jyecusch) [inline docs/troubleshooting/updating-nitric.mdx:9]: I wondering if a quick update the `doctor` command could help with this instead. So when you run:
`nitric doctor`
We could internally run:
- `docker pull nitricimages/membrane-local`
- `docker pull nitricimages/membrane-aws`
- `docker pull nitricimages/membrane-gcp`
- `docker pull nitricimages/membrane-do`
- `docker pull nitricimages/membrane-azure`

**OR** (probably better)

We could just prune the contents of the `nitricimages` repo in the local docker environment, which would force the images to be pulled the next time they're used and avoid pulling images that aren't needed (like clouds that aren't in use)

File: docs/troubleshooting/updating-nitric.mdx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  - Note - If you encounter issues with the previous step(s) then prune docker with the following command `docker system prune -a`
+- Update your local store of templates `nitric templates:repos:add`
+- In your project(s) remove the templates folder which resides in .nitric
+- Create a new project for each of the templates which you are using `nitric make:service` this will update the .nitric folder with the appropriate templates.
```
Comment (jyecusch) [inline docs/troubleshooting/updating-nitric.mdx:13]: This could/should also be done by the `nitric doctor` command or perhaps a new command like `nitric repair`, which finds services in the nitric.yaml file and if the referenced `runtime` is missing in the local `.nitric` file, prompts to add them back from the template repo.

So then the steps in the docs would be:

1. delete `./nitric/templates`
2. run `nitric repair` and follow the prompts

Comment (jyecusch) [reply docs/troubleshooting/updating-nitric.mdx:13]: @tjholm thoughts?

**Review by jyecusch — CHANGES_REQUESTED** [review]

**Comment by jyecusch** [comment]
The doc.json file will need to be updated to reference these new files. @raksiv let's review tomorrow and get this closed

File: docs/troubleshooting/updating-nitric.mdx (hunk 4 above)
Comment (raksiv) [reply docs/troubleshooting/updating-nitric.mdx:9]: Breaking this out into a new issue against the CLI. 
We can change the docs to add this when complete.

File: docs/troubleshooting/updating-nitric.mdx (hunk 5 above)
Comment (raksiv) [reply docs/troubleshooting/updating-nitric.mdx:13]: Adding this to CLI ticket.

**Review by jyecusch — CHANGES_REQUESTED** [review]
